### ISSUE TITLE
remove unnecessary operation in long_compare()

### ISSUE THREAD

**Issue by b8e5a65c-bed4-4c5d-ba32-799f883ba638** [python/cpython#79877]
BPO | [35696](https://bugs.python.org/issue35696)
--- | :---
Nosy | @vstinner, @methane, @animalize, @sir-sigurd, @hongweipeng
PRs | <li>python/cpython#11473</li><li>python/cpython#11473</li><li>python/cpython#11473</li><li>python/cpython#16146</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2019-01-09.09:26:39.577>
labels = ['3.9']
title = 'remove unnecessary operation in long_compare()'
updated_at = <Date 2019-09-21.09:56:56.010>
user = 'https://github.com/animalize'
```

bugs.python.org fields:
```python
activity = <Date 2019-09-21.09:56:56.010>
actor = 'hongweipeng'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = []
creation = <Date 2019-01-09.09:26:39.577>
creator = 'malin'
dependencies = []
files = []
hgrepos = []
issue_num = 35696
keywords = ['patch', 'patch', 'patch']
message_count = 5.0
messages = ['333293', '352732', '352875', '352883', '352903']
nosy_count = 5.0
nosy_names = ['vstinner', 'methane', 'malin', 'sir-sigurd', 'hongweipeng']
pr_nums = ['11473', '11473', '11473', '16146']
priority = 'normal'
resolution = None
stage = 'resolved'
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue35696'
versions = ['Python 3.9']
```

</p></details>

**Comment by iritkatriel** [comment]
Is there anything left to do here?

**Comment by vstinner** [comment]
I looked at Git history and it seems like I fixed the compiler warning with commit 6314abcc08f5d0f3d3a915dc9455ea223fa65517:

```
commit 6314abcc08f5d0f3d3a915dc9455ea223fa65517
Author: Victor Stinner <vstinner@redhat.com>
Date:   Tue Oct 1 13:29:53 2019 +0200

    bpo-37802: Fix a compiler warning in longobject.c (GH-16517)
    
    bpo-37802, bpo-38321: Fix the following warnings:
    
        longobject.c(420): warning C4244: 'function': conversion from
        'unsigned __int64' to 'sdigit', possible loss of data
    
        longobject.c(428): warning C4267: 'function': conversion from
        'size_t' to 'sdigit', possible loss of data
```

Thanks for the reminder @iritkatriel, I closed the issue.
